### PR TITLE
Move XCM runtime configurations into their own files

### PR DESCRIPTION
**Pull request by KiChjang**
This is purely a refactoring of XCM config code.

Since misconfigured XCM settings easily causes attacks, I thought that it may be useful to move its configuration into its own file for easier auditing and maintainability.

### CODE REVIEW

**Comment by shawntabrizi** [comment]
We should do this for the runtime in general, across all the pallets.

**Review by shawntabrizi — APPROVED** [review]

File: runtime/kusama/src/xcm_config.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
```
Comment (kianenigma) [inline runtime/kusama/src/xcm_config.rs:1]:
```suggestion
// Copyright 2022 Parity Technologies (UK) Ltd.
```

File: runtime/polkadot/src/xcm_config.rs [hunk 2]
```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
```
Comment (kianenigma) [inline runtime/polkadot/src/xcm_config.rs:1]:
```suggestion
// Copyright 2022 Parity Technologies (UK) Ltd.
```

File: runtime/polkadot/src/xcm_config.rs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+	BackingToPlurality<
+		Origin,
+		pallet_collective::Origin<Runtime, CouncilCollective>,
+		CouncilBodyId,
+	>,
+	// And a usual Signed origin to be used in XCM as a corresponding AccountId32
+	SignedToAccountId32<Origin, AccountId, PolkadotNetwork>,
+);
+
+impl pallet_xcm::Config for Runtime {
```
Comment (kianenigma) [inline runtime/polkadot/src/xcm_config.rs:167]: I would personally rather have this block in the main runtime file, and all the other noisy `parameter_types!` in a separate file. This allows you to quickly skim the runtime configs with 1 layer of observability and no further. For example, you will know the _name_ of the type parameters, but not their value, and their name could already say a lot of about them.

But this format is also fine.

File: runtime/rococo/src/xcm_config.rs [hunk 4]
```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
```
Comment (kianenigma) [inline runtime/rococo/src/xcm_config.rs:1]:
```suggestion
// Copyright 2022 Parity Technologies (UK) Ltd.
```

Comment (kianenigma) [reply runtime/rococo/src/xcm_config.rs:1]: and the rest please.

**Review by kianenigma — APPROVED** [review]

File: runtime/polkadot/src/xcm_config.rs (hunk 3 above)
Comment (shawntabrizi) [reply runtime/polkadot/src/xcm_config.rs:167]: i actually think this content in a separate file is the more natural thing to do. I think that we should make files where configurations with one another are relevant, so all democracy / governance configs in one file. But keep separate things which are not related

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**